File: QTD.SPA/src/app/_DtoModels/Report/ReportFilterOption.ts
import { ReportFilterOptionParent } from "./ReportFilterOptionParent";
import { ReportFilterTableColumns } from "./ReportFilterTableColumns";

export class ReportFilterOption{
    id:number;
    value:string;
    name:string;
    active? :boolean
    filterOptionParents: Array<ReportFilterOptionParent>;
    filterTableColumns: Array<ReportFilterTableColumns>;
    constructor(filterName: string, filterValue: string ){
        this.name = filterName;
        this.value = filterValue;
     }
}

File: QTD2.Infrastructure/Model/Reports/ReportFilterOption.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Infrastructure.Model.Reports
{
    public class ReportFilterOption
    {
        public string Name { get; set; }
        public string Value { get; set; }
        public bool? Active { get; set; }
        public bool IsDefaultOrder { get; set; } = false;

        public List<ReportFilterOptionParent> FilterOptionParents { get; set; }
        public List<ReportFilterTableColumns> FilterTableColumns { get; set; }

        public ReportFilterOption()
        {

        }
    }

    public class ReportFilterOptionParent
    {
        public string Name { get; set; }
        public List<string> Values { get; set; }
        public bool IsCascade { get; set; } = false;
        public bool IsTableVisible { get; set; } = true;
        public string ControlType{ get; set; } = "Dropdown";
    }

    public class ReportFilterTableColumns
    {
        public string Name { get; set; }
        public List<string> Values { get; set; }
    }
}
